sr/comp/mixtape/magicq.py

```diff
@@ -1,32 +1,18 @@
-import socket
-import struct
-from typing import Tuple
+from typing import Union
 
+from pythonosc.udp_client import SimpleUDPClient
 
-class MagicqController:
-
-    def __init__(self, address: Tuple[str, int]) -> None:
-        self.address = address
-        self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
 
-    @staticmethod
-    def build_packet(data: bytes) -> bytes:
-        length = struct.pack('<H', len(data))
-        return b'CREP\0\0\0\0' + length + data
-
-    def send_command_once(self, command: str) -> None:
-        self.send_command(command, 1)
+class MagicqController:
 
-    def send_command(self, command: str, retries: int = 5) -> None:
-        packet = self.build_packet(command.encode('ascii'))
-        for _retry in range(retries):
-            self.socket.sendto(packet, self.address)
+    def __init__(self, host: str, port: int) -> None:
```

Comment:
Suggest we keep this passing the address as a single variable. I realise that the underlying `SimpleUDPClient` wants them separately, however they are kinda logically one entity so it's slightly neater to pass them together. (Usage of `SimpleUDPClient` as `SimpleUDPClient(*address)` should work fine and still be type-safe etc.)

The API-design thinking here is two-fold:
- consider what happens when another, unrelated, parameter is added to the call -- are the three values siblings, or are some of them related to each other differently
- another way to think about that is to ask what would happen if we were to remove the host or port arguments -- would we also want to remove the other one

Reply:
I disagree, this more verbose form makes it clear what the contents of the data required is.

Also consider if we were adding another argument to specify the lighting console it would come from the magicq config key. The additional argument would then be a sibling of the current arguments.